packages/drivers/odsp-driver/src/odspSummaryUploadManager.ts

```diff
@@ -0,0 +1,412 @@
+        let blobs = 0;
+
+        const keys = Object.keys(tree.tree);
+        for (const key of keys) {
+            const summaryObject = tree.tree[key];
+
+            let id: string | undefined;
+            let value: SnapshotTreeValue | undefined;
+            const currentPath = path === "" ? key : `${path}/${key}`;
+            switch (summaryObject.type) {
+                case api.SummaryType.Tree: {
+                    blobTreeDedupCachesLatest.treesPathToTree.set(currentPath, summaryObject);
+                    const result = await this.convertSummaryToSnapshotTree(
+                        parentHandle,
+                        summaryObject,
+                        blobTreeDedupCachesLatest,
+                        rootNodeName,
+                        currentPath,
+                        expanded);
+                    value = result.snapshotTree;
+                    reusedBlobs += result.reusedBlobs;
+                    blobs += result.blobs;
+                    break;
+                }
+                case api.SummaryType.Blob: {
+                    let hash: string | undefined;
+                    let cachedPath: string | undefined;
+                    // If we are expanding the handle, then the currentPath should exist in the cache as we will get the blob
+                    // hash from the cache.
+                    if (expanded) {
+                        hash = this.blobTreeDedupCaches.pathToBlobSha.get(currentPath);
+                        assert(hash !== undefined, "hash should be set here");
+                        cachedPath = this.blobTreeDedupCaches.blobShaToPath.get(hash);
+                        assert(cachedPath !== undefined, "path should be defined as path->sha mapping exists");
+                    } else {
+                        value = typeof summaryObject.content === "string"
+                        ? { content: summaryObject.content, encoding: "utf-8" }
+                        : { content: Uint8ArrayToString(summaryObject.content, "base64"), encoding: "base64" };
+                        hash = await hashFile(IsoBuffer.from(value.content, value.encoding));
+                    }
+
+                    assert(hash !== undefined, "hash should be set!");
+                    // If the cache has the hash of the blob and handle of last summary is also present, then use that
+                    // cached path for the given blob. Also update the caches for future use.
+                    if (cachedPath === undefined || parentHandle === undefined) {
+                        blobs++;
+                        blobTreeDedupCachesLatest.blobShaToPath.set(hash, currentPath);
+                        blobTreeDedupCachesLatest.pathToBlobSha.set(currentPath, hash);
+                    } else {
+                        reusedBlobs++;
+                        id = `${parentHandle}/${cachedPath}`;
+                        blobTreeDedupCachesLatest.blobShaToPath.set(hash, currentPath);
```

Comment:
these two statements around blobTreeDedupCachesLatest are the same in both branches of IF - can you please move them out of IF?

Reply:
done